### PR TITLE
Feat/apply Easy Days to more than one date

### PR DESCRIPTION
**Pull request by L-M-Sherlock**
image

Patch: [fsrs4anki-helper.ankiaddon.zip](https://github.com/open-spaced-repetition/fsrs4anki-helper/files/14048948/fsrs4anki-helper.ankiaddon.zip)

@user1823, give it a try.

### CODE REVIEW

**Comment by user1823** [comment]
If I want to apply easy days to a single date, with this implementation, I have to first click `Add` and then `Apply`.
In my opinion, clicking `Apply` should run the function using the date entered within the box and `Add` should be used to add more dates.

**Comment by user1823** [comment]
Also, can you change the date format in the input box to ISO 8601 standard (YYYY-MM-DD)? 
I can't change the date to anything less than 25 unless I change the month first. (Anki is trying to prevent me from entering past dates.) In the ISO standard, the month comes first. So, the user will change the month first and the issue will not arise. In addition, the ISO standard is less confusing.

**Comment by L-M-Sherlock** [comment]
> In my opinion, clicking `Apply` should run the function using the date entered within the box and `Add` should be used to add more dates.

I'm afraid that it would induce confusion. Some users likely don't know the date in the DateEditBox will also be taken into account.

> Also, can you change the date format in the input box to ISO 8601 standard (YYYY-MM-DD)?

Do you mean a text input box? I don't want to deal with invalid user inputs.

> Otherwise, you can remove the following line and add a check after the user has pressed `Add` or `Apply`.

OK. I prefer this way.

**Comment by L-M-Sherlock** [comment]
patch: [fsrs4anki-helper.ankiaddon.zip](https://github.com/open-spaced-repetition/fsrs4anki-helper/files/14049384/fsrs4anki-helper.ankiaddon.zip)

File: schedule/easy_days.py [hunk 1]
```diff
@@ -73,32 +75,50 @@ def initUi(self):
         self.dateLabel.setText("Select The Specific Date")
```
Comment (user1823) [inline schedule/easy_days.py]:
```suggestion
        self.dateLabel.setText("Select the Date(s)")
```

File: schedule/easy_days.py [hunk 2]
```diff
@@ -73,32 +75,50 @@ def initUi(self):
...
         self.dateEdit = QDateEdit()
         self.dateEdit.setDateTime(QDateTime.currentDateTime())
-        self.dateEdit.setMinimumDate(QDate.currentDate())
 
-        self.applyEasyDayBtn = QPushButton("Apply Easy Day")
+        self.addDateBtn = QPushButton("Add the Selected Specific Date")
```
Comment (user1823) [inline schedule/easy_days.py:79]:
```suggestion
        self.addDateBtn = QPushButton("Add the Selected Date")
```

File: schedule/easy_days.py [hunk 3]
```diff
@@ -73,32 +75,50 @@ def initUi(self):
...
             tooltip("Please enable load balance first")
             return
-        specific_date = self.dateEdit.date().toPyDate()
-        current_date = datetime.now().date()
-        day_offset = (specific_date - current_date).days
-        today = mw.col.sched.today
-        specific_due = today + day_offset
-        due_days = [specific_due]
+        if len(self.specific_dates) == 0:
+            tooltip("Please add specific dates first")
```
Comment (user1823) [inline schedule/easy_days.py:112]:
```suggestion
            tooltip("Please add the dates first.")
```

File: schedule/easy_days.py [hunk 4]
```diff
@@ -61,44 +62,63 @@ def auto_easy_days():
...
         self.setLayout(self.layout)
-        self.setWindowTitle("Easy Day for a specific date")
+        self.setWindowTitle("Easy Days for Specific Dates")
+
+    def addEventFunc(self):
+        specific_date = self.dateEdit.date().toPyDate()
+        if specific_date in self.specific_dates:
+            tooltip("This date has already been added")
+            return
+        self.specific_dates.append(specific_date)
```
Comment (user1823) [inline schedule/easy_days.py:100]:
```suggestion
        if specific_date < datetime.now().date():
            tooltip("Easy days can't be applied on past dates.")
            return
        self.specific_dates.append(specific_date)
```

**Pull request merged**